lstchain/irf/hdu_table.py

```diff
@@ -0,0 +1,323 @@
+    hdu_header["HDUCLAS1"] = "INDEX"
+    hdu_header["HDUCLAS2"] = "HDU"
+    hdu_header["TELESCOP"] = "CTA"
+    hdu_header["INSTRUME"] = "LST-1"
+
+    filename_hdu_table = fits_dir/'hdu-index.fits.gz'
+
+    hdu = fits.BinTableHDU(hdu_table, header=hdu_header, name='HDU INDEX')
+    hdu_list = fits.HDUList([fits.PrimaryHDU(), hdu])
+    hdu_list.writeto(filename_hdu_table, overwrite=True)
+
+    obs_table = vstack(obs_tables)
+
+    obs_header = hdu_header.copy()
+    obs_header["HDUCLAS2"] = "OBS"
+    obs_header['MJDREFI'] = event_table.meta['MJDREFI']
+    obs_header['MJDREFF'] = event_table.meta['MJDREFF']
+
+    filename_obs_table = fits_dir/'obs-index.fits.gz'
+
+    obs = fits.BinTableHDU(obs_table, header = obs_header, name='OBS INDEX')
+    hdu_list = fits.HDUList([fits.PrimaryHDU(), obs])
+    hdu_list.writeto(filename_obs_table, overwrite=True)
+
+    return
+
+def create_event_list(data, run_number, source_name, mode):
+    """
+    Create the event_list BinTableHDUs from the given data
+
+    Parameters
+    ----------
+        Data: DL2 data file
+                'astropy.table.QTable'
+        Run: Run number
+                Int
+        Source_name: Name of the source
+                Str
+        Mode: Observation mode
+                Str, eg: ON, OFF, WOBBLE
+    Returns
+    -------
+        Events HDU:  `astropy.io.fits.BinTableHDU`
+        GTI HDU:  `astropy.io.fits.BinTableHDU`
+        Pointing HDU:  `astropy.io.fits.BinTableHDU`
+    """
+    name=source_name
+
+    # Timing parameters
+    lam = 2800 #Average rate of triggered events, taken by hand for now
+    t_start = data['dragon_time'][0].value
+    t_stop = data['dragon_time'][-1].value
+    time = Time(data['dragon_time'], format='unix', scale="utc")
+    date_obs = time[0].to_value('iso', 'date')
+    obs_time = t_stop-t_start #All corrections excluded
+
+    #Position parameters
+    focal = 28 * u.m
+    pos_x = data['reco_src_x']
+    pos_y = data['reco_src_y']
+    pointing_alt = data['pointing_alt']
+    pointing_az = data['pointing_az']
+
+    coord = camera_to_altaz(pos_x = pos_x, pos_y=pos_y, focal = focal,
+                    pointing_alt = pointing_alt, pointing_az = pointing_az,
+                    obstime = time)
+    coord_pointing = camera_to_altaz(pos_x = 0 * u.m, pos_y=0 * u.m, focal = focal,
+                pointing_alt = pointing_alt[0], pointing_az = pointing_az[0],
+                obstime = time[0])
+
+    object_radec=SkyCoord.from_name(source_name)
+
+    ##########################################################################
+    ### Event table
+    event_table = QTable(
+            {
+                "EVENT_ID" : u.Quantity(data['event_id']),
+                "TIME" : u.Quantity(data['dragon_time']),
+                "RA" : u.Quantity(coord.icrs.ra.to(u.deg)),
+                "DEC" : u.Quantity(coord.icrs.dec.to(u.deg)),
+                "ENERGY" : u.Quantity(data['reco_energy'])
+            }
+        )
```

Comment:
I don't understand this part
The DL2 data must include the reconstructed position in the sky already (`reco_alt` and `reco_az`).
Is there a reason to reconstruct these coordinates again?

Reply:
I have amended this.